File: Sample/java/src/main/java/net/nend/sample/java/banner/BannerSampleActivity.java
package net.nend.sample.java.banner;

import android.os.Bundle;

import net.nend.sample.java.R;
import net.nend.sample.java.SimpleListActivity;
import net.nend.sample.java.banner.adjust.AdjustSizeActivity;
import net.nend.sample.java.banner.layoutpatterns.DialogActivity;
import net.nend.sample.java.banner.layoutpatterns.LayoutSampleActivity;
import net.nend.sample.java.banner.noxml.JavaCallActivity;
import net.nend.sample.java.banner.sizes.SizeSampleActivity;
import net.nend.sample.java.banner.xmllayout.XmlSampleActivity;

import java.util.ArrayList;
import java.util.List;

public class BannerSampleActivity extends SimpleListActivity {

    private static final List<String> SAMPLE_ACTIVITIES = new ArrayList<String>() {
        {
            add(XmlSampleActivity.class.getName());
            add(JavaCallActivity.class.getName());
            add(LayoutSampleActivity.class.getName());
            add(DialogActivity.class.getName());
            add(SizeSampleActivity.class.getName());
            add(AdjustSizeActivity.class.getName());
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        instantiateMenuListFragment(R.layout.activity_banner_sample, SAMPLE_ACTIVITIES);
    }
}

File: Sample/kotlin/src/main/java/net/nend/sample/kotlin/banner/BannerSampleActivity.kt
package net.nend.sample.kotlin.banner

import android.app.Activity
import android.app.ListActivity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.ListView
import net.nend.sample.kotlin.R

class BannerSampleActivity : ListActivity() {

    private enum class SampleType(val id: Int) {
        XML_SAMPLE(0),
        NO_XML_SAMPLE(1),
        LAYOUT_SAMPLE(2),
        DIALOG_SAMPLE(3),
        SIZE_SAMPLE(4),
        ADJUST_SIZE_SAMPLE(5);

        companion object {
            fun getType(id: Int) = SampleType.values().first { it.id == id }
        }

        fun startActivity(activity: Activity) {
            when (this) {
                XML_SAMPLE -> activity.startActivity(Intent(activity,
                        XmlSampleActivity::class.java))
                NO_XML_SAMPLE -> activity.startActivity(Intent(activity,
                        NoXmlLayoutActivity::class.java))
                LAYOUT_SAMPLE -> activity.startActivity(Intent(activity,
                        LayoutSampleActivity::class.java))
                DIALOG_SAMPLE -> activity.startActivity(Intent(activity,
                        DialogActivity::class.java))
                SIZE_SAMPLE -> activity.startActivity(Intent(activity,
                        SizeSampleActivity::class.java))
                ADJUST_SIZE_SAMPLE -> activity.startActivity(Intent(activity,
                        AdjustSizeActivity::class.java))
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_banner_sample)
    }

    override fun onListItemClick(l: ListView?, v: View?, position: Int, id: Long) {
        super.onListItemClick(l, v, position, id)
        SampleType.getType(position).startActivity(this)
    }

    companion object {
        const val BANNER_SPOT_ID_320_50 = 3174
        const val BANNER_API_KEY_320_50 = "c5cb8bc474345961c6e7a9778c947957ed8e1e4f"
    }
}